File: GameOfLife.py
import random
import os
import time

xDimension = 50
yDimension = 50
chanceOfSpawning = 25
delay = 0.1 #ms Delay time for each frame

class Cell:
    dead = ' \033[91m' + '-' + '\033[0m '  # \033[91m = Red 
    alive = ' \033[96m' + '0' + '\033[0m ' # \033[96m = Cyan
                                           # \033[0m = Reset color

def RunGame():
	
	iteration = 0

	gameField = {}

	gameIsOn = True

	while(gameIsOn):
		ClearScreen()
		currentString = ""
		nextField = {}
		for i in range(yDimension):
			for j in range(xDimension):
				if iteration == 0:
					currentString = InitialPrint(j, i, currentString, gameField, chanceOfSpawning)
					continue
				nextField[(j, i)] = EvaluateCell(j, i, gameField)
				currentString += nextField[(j, i)]
			currentString += "\n"

		if iteration > 0:
			gameField=nextField
		
		iteration += 1
		
		print("ITERATION: " + str(iteration))
		print("CURRENT STRING = \n" + currentString)
		

		time.sleep(delay)

def InitialPrint(xDimension, yDimension, currentString, gameField, chanceOfSpawning):
	
	gameField[(xDimension, yDimension)] = Cell.alive if random.randrange(0, 100) <= chanceOfSpawning else Cell.dead
	currentString += gameField[(xDimension, yDimension)]

	return currentString

def EvaluateCell(xDimension, yDimension, gameField): 
 
	neighbors = 0

	for i in range(3):
		yOffset = i - 1
		for j in range(3):
			xOffset = j - 1
			if xOffset == 0 and yOffset == 0:
				continue
			try:
				if gameField[(xDimension + xOffset, yDimension + yOffset)] == Cell.alive:
					neighbors += 1	
			except:
				# Out of bounds position
				continue

	isAlive = gameField[(xDimension, yDimension)] == Cell.alive
	if isAlive:
		# Check the rules of Conway's Game of Life
	    if neighbors < 2 or neighbors > 3:
	        return Cell.dead
	    else:
	        return Cell.alive
	else:
	    if neighbors == 3:
	        return Cell.alive
	    else:
	        return Cell.dead
			

def ClearScreen():
    # Check the operating system name
    if os.name == 'nt':
        # Command for Windows
        _ = os.system('cls')
    else:
        # Command for Linux, macOS, and others
        _ = os.system('clear')

RunGame()

File: GameOfLife.js
document.addEventListener("DOMContentLoaded", () => {
    var background = document.createElement("DIV");
    background.setAttribute("id", "bg");

    var bgPara = document.createElement("P");
    bgPara.setAttribute("id", "game");
    var bgText = document.createTextNode("");

    bgPara.appendChild(bgText)
    background.appendChild(bgPara);


    let xDimension = 50;
    let yDimension = 50;
    let chanceOfSpawning = 50;
    let delay = 100; // milliseconds
    let gameField = {};


    //Initialize field
    for (let i = 0; i < yDimension; i++) {
        for (let j = 0; j < xDimension; j++) {
            let cellPos = j.toString() + "," + i.toString();
            gameField[cellPos] = (Math.floor(Math.random() * 100)) < chanceOfSpawning ? " 0 " : "   ";
        }
    }

    const gameLoop = setInterval(() => {
        let currentString = "";
        let nextField = {};

        for (let i = 0; i < yDimension; i++) {
            for (let j = 0; j < xDimension; j++) {
                let cellPos = j.toString() + "," + i.toString();
                nextField[cellPos] = EvaluateCell(j, i, gameField);
                currentString += nextField[cellPos];
            }
            currentString += "\n";
        }

        gameField = nextField;
        
        bgText.textContent = currentString;// Writing the field

    }, delay);

    function EvaluateCell(xPos, yPos, gameField) {
        let aliveCell = " 0 ";
        let deadCell = "   ";
        let neighbors = 0;

        for (let i = 0; i < 3; i++) {
            let yOffset = i - 1;
            for (let j = 0; j < 3; j++) {
                let xOffset = j - 1;
                if (xOffset == 0 && yOffset == 0) continue;

                let cellPos = (xPos + xOffset).toString() + "," + (yPos + yOffset).toString();
                if (gameField[cellPos] == aliveCell) {
                    neighbors += 1;
                }
            }
        }

        let cellPos = xPos.toString() + "," + yPos.toString();
        let isAlive = gameField[cellPos] == aliveCell;

        if (isAlive) {
            return (neighbors < 2 || neighbors > 3) ? deadCell : aliveCell;
        } else {
            return neighbors == 3 ? aliveCell : deadCell;
        }
    }

    document.body.appendChild(background);
});